ravenframework/SupervisedLearning/DMD/DMD.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on July 21, 2024
+
+  @author: Andrea Alfonsi
+  Traditional Dynamic Mode Decomposition
+
+"""
+#Internal Modules (Lazy Importer)--------------------------------------------------------------------
+from ...utils.importerUtils import importModuleLazy
+#Internal Modules (Lazy Importer) End----------------------------------------------------------------
+
+#External Modules------------------------------------------------------------------------------------
+np = importModuleLazy("numpy")
+pydmd = importModuleLazy("pydmd")
+ezyrb = importModuleLazy("ezyrb")
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from ...SupervisedLearning.DMD import DMDBase
+from ...utils import utils
+from ...utils import InputData, InputTypes
+#Internal Modules End--------------------------------------------------------------------------------
+
+class DMD(DMDBase):
```

Comment:
I am not sure what is the scope of this DMD class

Reply:
it is the base DMD. (original formulation of the DMD method)